omod/src/main/java/org/openmrs/module/idgen/rest/resource/IdentifierSourceResource.java

```diff
@@ -257,6 +257,176 @@ else if(identifierSourceType.equals(Identifier_Pool)){
+                    }
+                }
+                Context.getService(IdentifierSourceService.class).saveIdentifierSource(source);
+                output.add("message", "Successfully uploaded reserved identifiers.");
+                return output;
+            }
+        }
+        if (batchSizeToUpload != null) {
+            Object sourceUuid = updateBody.get("sourceUuid");
+            if(sourceUuid != null && Context.getService(IdentifierSourceService.class).getIdentifierSourceByUuid(sourceUuid.toString()) != null){
+                IdentifierPool pool = (IdentifierPool) Context.getService(IdentifierSourceService.class).getIdentifierSourceByUuid(sourceUuid.toString());
+                Context.getService(IdentifierSourceService.class).addIdentifiersToPool(pool, Integer.parseInt(batchSizeToUpload.toString()));
+                output.add("message", "Successfully added batch " +batchSizeToUpload.toString()+ " from source.");
+                return output;
+            }
+        }
+        if (identifiersToUpload != null) {
+            Object sourceUuid = updateBody.get("sourceUuid");
+            List<String> ids = (ArrayList) identifiersToUpload;
+            if(sourceUuid != null && Context.getService(IdentifierSourceService.class).getIdentifierSourceByUuid(sourceUuid.toString()) != null){
+                IdentifierPool pool = (IdentifierPool) Context.getService(IdentifierSourceService.class).getIdentifierSourceByUuid(sourceUuid.toString());
+                Context.getService(IdentifierSourceService.class).addIdentifiersToPool(pool, ids);
+                output.add("message", "Successfully uploaded identifiers to pool.");
+                return output; 
+            }
+        }
+        if (generateIdentifiers != null) {
+            Object comment = updateBody.get("comment");
+            Object numberToGenerate = updateBody.get("numberToGenerate");
+            Object sourceUuid = updateBody.get("sourceUuid");
+            if (comment == null) {
+                comment = "Batch Export of " + numberToGenerate + " to file";
+            }
+            if(numberToGenerate == null){
+                numberToGenerate = 0;
+            }
+            if(sourceUuid != null && Context.getService(IdentifierSourceService.class).getIdentifierSourceByUuid(sourceUuid.toString()) != null){
+                IdentifierSource source = Context.getService(IdentifierSourceService.class).getIdentifierSourceByUuid(sourceUuid.toString());
+                List<String> batch = Context.getService(IdentifierSourceService.class).
+                        generateIdentifiers(source, Integer.parseInt(numberToGenerate.toString()), comment.toString());
+                output.add("identifiers", batch.toString());
+                return output;
+            }   
+        }
+        if(Context.getService(IdentifierSourceService.class).getIdentifierSourceByUuid(uuid) instanceof SequentialIdentifierGenerator){
```

Comment:
Is it necessary to call this twice? Context.getService(IdentifierSourceService.class).getIdentifierSourceByUuid(uuid)

Reply:
fixed